### ISSUE TITLE
Standalone agent for Kubernetes keeps crashing and restarting

### ISSUE THREAD

**Issue by akshay-saraswat** [elastic/integrations#1354]
I am following this guide to run the standalone agent on Kubernetes (GKE version : 1.19.9-gke.1900, machine : e2-medium, image :  cos_containerd) for metrics collection.
https://www.elastic.co/guide/en/fleet/current/running-on-kubernetes-standalone.html

I am consistently getting OOMKilled and CrashLoopBackOff for these agent installations on GKE.

![image](https://user-images.githubusercontent.com/14043218/126995492-45cda86a-0a6d-4a31-9656-9d2d6de37b5c.png)

**Comment by ChrsMark** [comment]
cc: @michaelKatsoulis

**Comment by MichaelKatsoulis** [comment]
@akshay-saraswat Which image version of elastic-agent are you using?

**Comment by ChrsMark** [comment]
@akshay-saraswat we tried to reproduce it (using 7.13.3) with @MichaelKatsoulis on GKE (same as yours) and indeed we got the same `OOMKilled` failure. We had to increase the memory limits at https://github.com/elastic/beats/blob/e8ea14bd206f205a30dcdc843e336f7d83ec1620/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml#L494. Increasing to `400mi` fixed the issue however the cluster we were operating was just a basic one with very few pods so I guess the memory limit would need to be even higher in actual deployments. This is what we observed:
```
kubectl -n kube-system top pod fluentbit-gke-b79t8                                                                               
NAME                  CPU(cores)   MEMORY(bytes)   
elastic-agent-45m8r   19m          348Mi 
```

I would say that we need to change the default limits to something higher for our proposed manifests. Similar issues were reported at https://github.com/elastic/observability-test-environments/issues/1043.

@exekias do you know what process we followed  so as to identify the limits for Beats in the past? Maybe Agent team can provide an estimation of how much resources Agent would demand? I think this is not a k8s specific issue but it just shows up in k8s where the limits are strict.

**Comment by exekias** [comment]
> @exekias do you know what process we followed so as to identify the limits for Beats in the past? Maybe Agent team can provide an estimation of how much resources Agent would demand? I think this is not a k8s specific issue but it just shows up in k8s where the limits are strict.

We didn't benchmark this but used our previous experiences in FB/MB memory usages, we don't have that for Agent yet. Take into account that for bigger clusters agents are still in charge of their own node, so I don't expect their usage to grow a lot. 400MB seems good having those readings intro account. If this creates issues we will see them soon.

One question from my side: Is this the leader agent or all of them use about the same amount of memory?

**Comment by ChrsMark** [comment]
To be honest we didn't check if the specific Agent was holding the leadership lock, however it was not the only one crashing with OOMKilled. Also the testing cluster was fairly new with no additional resources running, so I wouldn't expect that cluster-wide metrics could be that much to overload the leader (in that case).

I will open a PR to increase the limits to `500mi` for now.

**Comment by ChrsMark** [comment]
Closing this since the OOM should be resolved by elastic/beats#27153. Let's keep tracking any open issues at https://github.com/elastic/observability-test-environments/issues/1043.
